py/mpconfig.h

```diff
@@ -739,4 +739,8 @@ typedef double mp_float_t;
 #define MP_UNLIKELY(x) __builtin_expect((x), 0)
 #endif
 
+// Use to refer to extern structs of which actual type is not known.
+// For instance 'extern struct _dummy_t mp_sys_stdin_obj;'
+struct _dummy_t;
```

Comment:
I agree we don't need the declaration in a header file.  Even gcc with -Wpedandic doesn't complain about it.

I would just name it `struct _mp_dummy_t` to be consistent with other names, and highlight the fact that it doesn't matter what it is and that a given port might have it as one type, another port as another type.